torch/distributed/_shard/sharding_spec/api.py

```diff
@@ -110,3 +110,62 @@ def __post_init__(self):
             rank = len(shard.shard_offsets)
 
         validate_non_overlapping_shards_metadata(self.shards)
+
+
+def _infer_sharding_spec_from_shards_metadata(shards_metadata):
+    """
+    Infer the sharding spec from the metadata of each shard of a ShardedTensor.
+    If the tensor is sharded only on one dimension, we then assume it's chunk sharded.
+    Otherwise, we assume it's enum sharded.
+
+    Args:
+        shards_metadata (List[ShardMetadata]): List of Metadata of local shards.
+
+    Returns:
+        A :class:`torch.distributed._shard.sharding_spec.ShardingSpec` object of sharding
+            spec for one sharded tensor.
+    """
+    placements = []
+    chunk_sharding_dim = None
+    chunk_offset_list = []
+    shard_size_list = []
+    # collect local shard metadatas from the global sharded_tensor_metadata
+    for shard_metadata in shards_metadata:  # type: ignore[attr-defined]
+        placements.append(shard_metadata.placement)
+        local_offsets = shard_metadata.shard_offsets
+        chunk_offset_list.append(sum(local_offsets))
+        shard_size_list.append(shard_metadata.shard_sizes)
+        shard_dims = [idx for idx, e in enumerate(local_offsets) if e != 0]
+        if len(shard_dims) == 0:
+            continue
+        if len(shard_dims) != 1:
+            break
+        if not chunk_sharding_dim:
+            chunk_sharding_dim = shard_dims[0]
```

Comment:
is this saying that if we have `len(shard_dims) == 1` for a single metadata, we can assume this is a chunkshardingspec?

it might not always be the case i guess? i.e. a sharded tensor with global size: (10, 10)
Shard 0: offset: [0, 0], 
Shard 1: offset: [5, 0]
Shard 2: offset: [0, 5]
Shard 3: offset: [5, 5]

This is actually not a chunkShardingSpec but rather a EnumerableShardingSpec?

Reply:
So, for this case. We will know it's an enum because 
For [5, 5], len(shard_dims) != 1.
For [5, 0] and then [0, 5], chunk_sharding_dim != shard_dims[0]. 

No matter what sequence you have in the sharding spec, we will know it's a enum sharding spec.

Reply:
I see, yeah make sense.